luthier/config/src/utils/defaultingUtils.lua

```diff
@@ -68,4 +71,17 @@ function M.logical_table_defaulting(logical_table_name, logical_table)
     logi_config.dateTimeZone = logi_config.dateTimeZone or M.DEFAULT_LOGICAL_DATE_TIME_ZONE
     return logi_config
 end
+
+function M.metric_defaulting(metric_name, metric)
+    local metric_config = misc.shallow_copy(metric)
+    metric_config.longName = metric_config.longName or metric_name
+    metric_config.description = metric_config.description or metric_config.longName   -- this line must follow the previous one
+    metric_config.category = metric_config.category or M.DEFAULT_METRIC_CATEGORY
+    metric_config.dataType = metric_config.dataType or M.DEFAULT_METRIC_DATA_TYPE
+    --- the following unification is no longer necessary in my (Gabriel's) opinion.
+    --- the discrepancy can be handled by two types of Factories: Aggregation and PostAggregation
```

Comment:
Why is this no longer necessary? Using "dependencies" to refer to a thing in Druid is weird and confusing. We set people up to view `dependencies` as "other Fili metrics this metric depends on." So we went with `druidMetric` as an alias for it that people can use for simple metrics.

Besides, is it necessarily the case than Aggregation will never depend on other Fili metrics? I'm not convinced it is.

Reply:
It is not necessary to unify `druidMetric` and `dependencies` into a so-called `dependencyMetricNames`. 
in the current json output, there will be `druidMetric` for AggregationMetricFactory to read, and a `dependencies` for PostAggregationMetricFactory to read, respectively.

Reply:
For your reference, this is not changing the interface we expose to users. the `metrics.lua` is still the same.

Reply:
I don't think this is a good use of the word `type.` I think when people ask "what type of metric something is" they mean "longSum" or "complicated formula." They don't
think "metric in Fili that relies on other metrics" vs "metric in Fili that's just a simple aggregation of a Druid metric." So I think it would be better to rename `maker` to `type`,
and drop this field completely.  Requiring people to add another field whose sole purpose is to signal whether this is a simple or complex metric is ceremony. It's already unambiguous
whether a metric is simple or not: If it has the `druidMetric` field, then it's simple. If it has the `dependencies` field, then it's complex. The unification is purely an implementation
detail so that Fili doesn't have to care whether a metric is simple or not. If we want to push that decision into the Java code, then that's fine. But the decision should turn on
the existence of the `druidMetric` field.

Reply:
transitively fixed. Now everything is defaulted to have a type `default`. Taking away this field in the end json file is not easy, as we have configured the java component to automatically look for the `FACTORY_KEY` i.e. `type` and find out which factory we need to invoke.